spec/acceptance/should_create_task_spec.rb

```diff
@@ -67,4 +77,177 @@
+    result = on(host, query_cmd)
+
+    # Verify the task is running under the correct user
+    result.stdout.match?(@username)
+  end
+
+  it "Should create a task with a username and password: win32_taskscheduler" do
+    pp = <<-MANIFEST
+    scheduled_task {'#{@taskname}':
+      ensure        => present,
+      command       => 'c:\\\\windows\\\\system32\\\\notepad.exe',
+      arguments     => "foo bar baz",
+      working_dir   => 'c:\\\\windows',
+      user          => '#{@username}',
+      password      => '#{@password}',
+      trigger       => {
+        schedule   => daily,
+        start_time => '12:00',
+      },
+      provider      => 'win32_taskscheduler'
+    }
+    MANIFEST
+    execute_manifest(pp, :catch_failures => true)
+
+    # Ensure it's idempotent
+    execute_manifest(pp, :catch_changes  => true)
+
+    # Verify the task exists
+    query_cmd = "schtasks.exe /query /v /fo list /tn #{@taskname}"
+    result = on(host, query_cmd)
+
+    # Verify the task is running under the correct user
+    result.stdout.match?(@username)
+  end
+
+  it "Should update a task's credentials: win32_taskscheduler" do
+    pp = <<-MANIFEST
+    scheduled_task {'#{@taskname}':
+      ensure        => present,
+      command       => 'c:\\\\windows\\\\system32\\\\notepad.exe',
+      arguments     => "foo bar baz",
+      working_dir   => 'c:\\\\windows',
+      user          => '#{@username}',
+      password      => '#{@password}',
+      trigger       => {
+        schedule   => daily,
+        start_time => '12:00',
+      },
+      provider      => 'win32_taskscheduler'
+    }
+    MANIFEST
+    execute_manifest(pp, :catch_failures => true)
+
+    # Ensure it's idempotent
+    execute_manifest(pp, :catch_changes  => true)
+
+    # Verify the task exists
+    query_cmd = "schtasks.exe /query /v /fo list /tn #{@taskname}"
+    result = on(host, query_cmd)
+
+    # Verify the task is running under the correct user
+    result.stdout.match?(@username)
+
+    # Get a new username and password
+    username_2, password_2 = add_test_user(host)
+
+    pp = <<-MANIFEST
+    scheduled_task {'#{@taskname}':
+      ensure        => present,
+      compatibility => 1,
+      command       => 'c:\\\\windows\\\\system32\\\\notepad.exe',
+      arguments     => "foo bar baz",
+      working_dir   => 'c:\\\\windows',
+      user          => '#{@username2}',
+      password      => '#{@password2}',
+      trigger       => {
+        schedule   => daily,
+        start_time => '12:00',
+      },
+      provider      => 'win32_taskscheduler'
+    }
+    MANIFEST
+    execute_manifest(pp, :catch_failures => true)
+
+    # Ensure it's idempotent
+    execute_manifest(pp, :catch_changes  => true)
+
+    # Verify the task exists
+    query_cmd = "schtasks.exe /query /v /fo list /tn #{@taskname}"
+    result = on(host, query_cmd)
+
+    # Verify the task is running under the correct user
+    result.stdout.match?(@username)
```

Comment:
Shouldn't this be username2 since we updated the task to the new creds of username2 and password2?

Reply:
It switches to the second username later in the file.